lang_js/analyze/transpile_js.ml

```diff
@@ -146,7 +146,21 @@ let compile_pattern (expr, fname, fpname) varname pat =
                     pname)
        in
        var_of_simple_pattern (expr, fname) init_builder pat
-     | _ -> failwith "TODO: PatObj pattern not handled"
+     | C.PatDots (tok, pat) ->
+       (* Make a fake "..." slice here:
+        *   var {...a} = b --> var a = b["..."]
+        *)
+       let init_builder (_name, _tok) =
+         A.ArrAccess (
+           A.Id (varname, ref A.NotResolved),
+           A.String ("...", tok)
+         )
+       in
+       var_of_simple_pattern (expr, fname) init_builder pat
+     | C.PatNest (_, _) ->
+       failwith "TODO: nesting inside object pattern not handled"
+     | _ ->
+       failwith "Unexpected pattern PatObj or PatArr found inside PatObj"
```

Comment:
Instead of a `_` catch-all, I would list all the possible cases (`PatObj _ | PatArr _`). It's a little more tedious to write but it guarantees an error message when a case is added to the type definition later and we forget to handle it here.

Reply:
Makes sense 👍